src/types/block_certificates.rs

```diff
@@ -0,0 +1,120 @@
+    /// [`BlockHash`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub block_hash:          BlockHash,
+    /// [`Round`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub round:               Round,
+    /// [`Epoch`] of the block that this [`QuorumCertificate`]
+    /// certifies.
+    pub epoch:               Epoch,
+    /// The aggregate signature on the block identified by the
+    /// `block_hash` which serves as a proof that the block
+    /// was accepted on the chain by a quorum of the finalization
+    /// committee.
+    pub aggregate_signature: QuorumSignature,
+    /// The baker ids of the finalizers that formed
+    /// the `aggregate_signature`.
+    /// Note that the baker ids are sorted in ascending order and are
+    /// distinct.
+    pub signatories:         Vec<BakerId>,
+}
+
+/// A map from a [`Round`] to the set of finalizers
+/// (identified by their [`BakerId`]) which signed off
+/// in the round.
+#[derive(Debug)]
+pub struct FinalizerRound {
+    /// The round which was signed off.
+    pub round:      Round,
+    /// The set of finalizers who signed off.
+    /// (identified by their [`BakerId`])
+    /// Note that the baker ids are sorted in ascending order and are
+    /// distinct.
+    pub finalizers: Vec<BakerId>,
+}
+
+/// An aggregate signature on a [`TimeoutCertificate`] created
+/// by members of the finalization committee.
+#[derive(concordium_base::common::Serialize, Clone, Debug, PartialEq)]
+pub struct TimeoutSignature(
+    pub concordium_base::aggregate_sig::Signature<concordium_base::base::AggregateSigPairing>,
+);
+
+/// The timeout certificate serves as a proof that no block
+/// was created and/or distributed to the network in time.
+/// The [`TimeoutCertificate`] makes it possible for the consenssus protocol
+/// to advance to the following round, thus giving (possibly) another baker
+/// the chance to bake a block.
+#[derive(Debug)]
+pub struct TimeoutCertificate {
+    /// The round that timed out.
+    pub round:                  Round,
+    /// The minimum epoch of which signatures are included in
+    /// the signature for the certificate.
+    pub min_epoch:              Epoch,
+    /// The rounds of which finalizers have their best quorum
+    /// certificates in the [`Epoch`] `min_epoch`.
+    pub qc_rounds_first_epoch:  Vec<FinalizerRound>,
+    /// The rounds of which finalizers have their best quorum
+    /// certificates in the [`Epoch`] `min_epoch` + 1.
+    pub qc_rounds_second_epoch: Vec<FinalizerRound>,
+    /// The aggregate signature by the finalization committee which
+    /// serves as a proof that the [`Round`] timed out, hence
+    /// no block was added to the chain.
+    pub aggregate_signature:    TimeoutSignature,
+}
+
+/// A proof that establishes that a block of a
+/// [`EpochFinalizationEntry`] is an
+/// immediate successor of the finalized block.
+#[derive(Debug)]
+pub struct SuccessorProof(pub [u8; 32]);
```

Comment:
This is a hash, so we should define it analogously to other hashes, e.g., block hash https://github.com/Concordium/concordium-base/blob/32a0350cb880b62a96f9a73a902fb7b2fd81258c/rust-src/concordium_base/src/hashes.rs#L44C14-L44C14

That way you get a lot of instances such as JSON, FromStr, Display, etc, for free.